### PR TITLE
feat: add security warnings to sandboxed renderers

### PR DESCRIPTION
**Pull request by miniak**
##### Description of Change
Security warnings are not logged in sandboxed renderers. This PR fixes that + refactors the implementation not to rely on the `remote` module, as https://github.com/electron/electron/pull/13028 adds an option to disable it.

##### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes
Notes: Security warnings added to sandboxed renderers.

### CODE REVIEW

File: lib/browser/rpc-server.js [hunk 1]
```diff
@@ -457,3 +457,11 @@ ipcMain.on('ELECTRON_BROWSER_SANDBOX_LOAD', function (event) {
     }
   }
 })
+
+ipcMain.on('ELECTRON_BROWSER_GET_LAST_WEB_PREFERENCES', function (event) {
+  try {
+    event.returnValue = event.sender.getLastWebPreferences()
+  } catch (error) {
```
Comment (alexeykuzmin) [inline lib/browser/rpc-server.js]: Should this error be logged?

File: lib/renderer/security-warnings.js [hunk 2]
```diff
@@ -77,204 +77,200 @@ const isUnsafeEvalEnabled = function () {
...
+    if (shouldLogSecurityWarnings()) {
+      const { ipcRenderer } = require('electron')
+      const webPreferences = ipcRenderer.sendSync('ELECTRON_BROWSER_GET_LAST_WEB_PREFERENCES')
+      logSecurityWarnings(webPreferences, nodeIntegration)
     }
-  }
 
-  // Currently missing since we can't easily programmatically check for it:
-  //   #12WebViews: Verify the options and params of all `<webview>` tags
+    window.removeEventListener('load', loadHandler)
```
Comment (alexeykuzmin) [inline lib/renderer/security-warnings.js]: You could simply add a third argument to the `.addEventListener()` call:
```js
window.addEventListener('load', function () {...}, {once: true})
```

Comment (miniak) [reply lib/renderer/security-warnings.js]: done

File: lib/browser/rpc-server.js (hunk 1 above)
Comment (miniak) [reply lib/browser/rpc-server.js]: it was not logged before. I can either log it here or transfer it (use `errorUtils.serialize`) and log it on the renderer process side (use `errorUtils.deserialize`)

Comment (miniak) [reply lib/browser/rpc-server.js]: opted for the 2nd option

File: lib/renderer/init.js [hunk 3]
```diff
@@ -27,23 +27,11 @@ v8Util.setHiddenValue(global, 'ipc', new events.EventEmitter())
...
-  warnAboutInsecureResources,
-  warnAboutInsecureCSP,
-  warnAboutAllowedPopups,
-  shouldLogSecurityWarnings
-} = require('@electron/internal/renderer/security-warnings')
-
 require('@electron/internal/renderer/web-frame-init')()
 
 // Process command line arguments.
-let nodeIntegration = 'false'
```
Comment (MarshallOfSound) [inline lib/renderer/init.js]: This change seems unrelated and concerns me slightly, I agree this variable should probable be a boolean rather than a string representation of a boolean but I think this change needs more scrutiny and doesn't belong in this PR.  Can you extract this change into a different PR.

**Comment by MarshallOfSound** [comment]
Looking good, just one extraction request

File: lib/renderer/init.js (hunk 3 above)
Comment (miniak) [reply lib/renderer/init.js]: will do that

Comment (miniak) [reply lib/renderer/init.js]: @MarshallOfSound done
